### DISCUSSION TITLE
How to select latest non-beta Xcode?

### DISCUSSION THREAD

**Discussion by Cyberbeni** [community/community#26214] [Actions]
<p>I’m using this, which selects Xcode_11.3.app but printing the xcodebuild version reveals that it’s the beta build.</p>
<pre><code>- name: Select latest Xcode
      run: |
        sudo xcode-select --switch "/Applications/$(ls "/Applications" | grep "Xcode_[\.0-9]*app" | sort -V | tail -1)/Contents/Developer"
        xcodebuild -version
</code></pre>

**Comment by weide-zhou** [comment]
<p>Github-hosted runner support below xcode by default.</p>
<p>Please use below command to get the 11.2.1 version xcode.</p>
<pre><code>sudo xcode-select --switch "/Applications/$(ls "/Applications" | grep "Xcode_[\.0-9]*app" | sort -V | tail -2 | head -1)/Contents/Developer"
xcodebuild -version
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/f8c814134980ca62731559d3f649ffaff136287d.png" data-download-href="/uploads/short-url/zuP4SyfMNY39egji6h9II7QTVJb.png?dl=1" title="xcodeversionongithubrunner.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181096093-bff1a4d5-9e35-4e44-98c3-ee97f95630e0.png" alt="xcodeversionongithubrunner.png" data-base62-sha1="zuP4SyfMNY39egji6h9II7QTVJb" width="690" height="350" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/f8c814134980ca62731559d3f649ffaff136287d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">xcodeversionongithubrunner.png</span><span class="informations">744×378 11.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/06055743ec50707a15cc2e5394e51d180d7964b8.png" data-download-href="/uploads/short-url/Rgj4r35KVgch47YJ9v0oJIOrbi.png?dl=1" title="xcodeselect.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181096094-6acc479b-c0a5-446a-9a06-a3dab7660b8a.png" alt="xcodeselect.png" data-base62-sha1="Rgj4r35KVgch47YJ9v0oJIOrbi" width="690" height="92" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/06055743ec50707a15cc2e5394e51d180d7964b8_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">xcodeselect.png</span><span class="informations">1139×153 6.25 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Cyberbeni** [comment]
<p>Sadly that only works when there is a beta version available. If the latest one is not beta then that will select the previous version.</p>
<p>I guess I have to look for Xcode_[.0-9]*_.*app and then select the last -(x+1)th Xcode version based on the number of matches.</p>

**Comment by Cyberbeni** [comment, marked as answer]
<p>Ah, realized how the symlinks are structured, so basically we want to only grep the non-symlinks and then betas won’t be included after the grep:</p>
<pre><code>sudo xcode-select --switch "$(find /Applications -mindepth 1 -maxdepth 1 ! -type l | grep "Xcode_[\.0-9]*app" | sort -V | tail -1)/Contents/Developer"
</code></pre>
